### ISSUE TITLE
[v438] Very odd glitch with pen drawing - hard to describe, see linked project...

### ISSUE THREAD

**Issue by TheLogFather** [LLK/scratch-flash#855]
Try out https://scratch.mit.edu/projects/10143939/
Note the orange spaceship jet trails, which last a second or two then disappear.
Your own ship's jet trail never appears.
EDIT: just noticed that the other ship's jet trails disappear at the point when you start moving forward (press W), i.e. when your own jet trail should appear...

After playing a while longer, something goes very wrong... it starts to get very laggy, and large circles can sometimes appear on screen.

At the moment, I have no idea where this problem has come from - note that Stage3D (or not) makes no difference. Will have a go at tracking it down a bit later...

**Comment by 2jour** [comment]
Hi @TheLogFather do you notice different behaviour between 438 and 437? I just clicked on the green flag

https://www.dropbox.com/s/k4rqccshqmx4sxf/v437.mp4?dl=0  v437

https://www.dropbox.com/s/ej22otp4y4f279k/v438.mp4?dl=0   v438

**Comment by TheLogFather** [comment]
You need to start moving around using WASD and arrow keys. In particular (and oddly), it only starts to go wrong once it tries to draw the jet-trails for your own ship, and that happens once you move forwards (with the W key).

The first thing to note is that the jet trail for your own ship never appears. Secondly, the jet-trails for the other ships disappear soon after you move forwards. Thirdly, it soon starts to get very laggy, even freezing. Fourthly, occasionally, there are large dots that appear in localised places on the screen - as if pen-down/pen-up with maximum size pen (and they're sometimes semi-transparent, too).

This glitch seems really strange, and I don't know what to make of it at the moment - I can't understand why so much other pen-drawing isn't affected (not only in this project, but so many others too) - why should this specific drawing in this project be so different...? :/

**Comment by TheLogFather** [comment]
I finally got to the bottom of this problem...

It turned out to be a bug in the scripting, which was causing it to reference item zero of the trails info list. That gave back a blank result, which turned out to be harmless for v437, but not for v438 because of the change of behaviour in the comparison functions, i.e. because of https://github.com/LLK/scratch-flash/issues/857 (it then caused chaos throughout the trails list, starting from the first one, which is your own ship, propagating through to others once that gets messed - hence the reason they all disappear soon after, or get some totally wacky values sometimes, thus causing the large circles to appear in odd places when it tries to draw them.)

It mean that v438.1 makes the project appear to run fine again...

Closing!
